OpenSim/Simulation/Model/Model.cpp

```diff
@@ -2220,8 +2220,31 @@ SimTK::Vec3 Model::calcMassCenterVelocity(const SimTK::State &s) const
 SimTK::Vec3 Model::calcMassCenterAcceleration(const SimTK::State &s) const
 {
     getMultibodySystem().realize(s, Stage::Acceleration);
-    return getMatterSubsystem().calcSystemMassCenterAccelerationInGround(s);    
+    return getMatterSubsystem().calcSystemMassCenterAccelerationInGround(s);
 }
+/**
+ * Return the spatial momentum about the system mass center expressed in Ground.
+ *
+ */
+SimTK::SpatialVec Model::calcMomentum(const SimTK::State &s) const
+{
+    return getMatterSubsystem().calcSystemCentralMomentum(s);
```

Comment:
Following the examples above (e.g., `Model::calcMassCenterAcceleration()`), shouldn't this line be preceded by
``` cpp
getMultibodySystem().realize(s, Stage::Velocity);
```

Reply:
(I think the `realize()` calls should be here since an API user might call the method directly. IIRC, calling `realize()` isn't [supposed to be] expensive if the State has already been realized to the required Stage.)

Reply:
I thought it was weird these calls were inside the calc-COM methods, but I think you're right, the `Stage` is probably cached and `realize` won't re-realize if it is. (I also tried taking out the realize calls in the calc-COM methods and it broke some tests, so there's that.)